src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,76 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { createComment } from '../api/comments';
+import { Post } from '../types/Post';
+import { useComments } from './Contexts/CommentsContext';
+
+type Props = {
+  selectedPost: Post,
+};
+
+export const NewCommentForm: React.FC<Props> = ({ selectedPost }) => {
+  const { setComments } = useComments();
+
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [commentText, setCommentText] = useState('');
```

Comment:
If we want to have a managed state for one entity with simple inputs, it is better to keep everything in one state.